docs: modify docs on doc testing to reflect code snippet execution

magurotuna opened the pull request:
We landed "true" doc testing in https://github.com/denoland/deno/pull/25220. With this change, `deno test --doc` will actually _run_ code snippets written in JSDoc or markdown files, as well as type checking. This commit updates the corresponding doc pages to follow this change.

Preview of the updated pages:

- https://deno-docs--magurotuna-improved-doc-te.deno.dev/runtime/fundamentals/testing/#documentation-tests
- https://deno-docs--magurotuna-improved-doc-te.deno.dev/runtime/fundamentals/typescript/#type-checking
- https://deno-docs--magurotuna-improved-doc-te.deno.dev/runtime/reference/cli/check/
- https://deno-docs--magurotuna-improved-doc-te.deno.dev/runtime/reference/cli/test/

thisisjofrank commented on runtime/fundamentals/testing.md:
```diff
@@ -339,44 +339,107 @@ are up-to-date and functional.
-using the `import` keyword followed by the relative path to the module. This
-allows you to test the code in the example block in the context of the module
-being documented.
+Looking at the generated test code above carefully, you will notice that it
```
```suggestion
Looking at the generated test code above, you will notice that it
```

thisisjofrank commented on runtime/fundamentals/testing.md:
```diff
@@ -339,44 +339,107 @@ are up-to-date and functional.
-allows you to test the code in the example block in the context of the module
-being documented.
+Looking at the generated test code above carefully, you will notice that it
+includes the `import` statement to import the `add` fucntion even though the
```
```suggestion
includes the `import` statement to import the `add` function even though the
```

thisisjofrank commented on runtime/fundamentals/testing.md:
```diff
@@ -339,44 +339,107 @@ are up-to-date and functional.
-being documented.
+Looking at the generated test code above carefully, you will notice that it
+includes the `import` statement to import the `add` fucntion even though the
+original code block does not have it. Briefly, all items imported from the
+module being documented are automatically imported into the generated test code
+with the same name.
```
```suggestion
original code block does not have it. When documenting a module, any imported items are automatically included in the generated test code using with the same name.
```

magurotuna replied:
Oh I now realized that the original sentence was wrong. What I meant is "any exported items from the module will automatically get imported in the generated test code using the same name". I'll fix the sentence anyway following your suggestion 👍

magurotuna replied:
Fixed [610af29](https://github.com/denoland/docs/pull/860/commits/610af29aeb9a24a412ff07a14bbf7df4867c0dd6)

magurotuna commented:
@thisisjofrank Thanks for valuable review comments! I addressed them, and I realized I had forgotten to mention how to skip code blocks. This is now added in [7505cf4](https://github.com/denoland/docs/pull/860/commits/7505cf467f2eecb0e900c137d7fb5dbb1ec55e15). Can you give a final review?

thisisjofrank reviewed (approved):
LGTM!
